rero/rero-ils
# users: add new fields, move fields from the patron

* Replaces the phone field by home, business, mobile, other phone fields
  in the user data model.
* Adds country and gender fields in the user data model.
* Adds 3 invenio-userprofiles configuration to specify the list of
  countries, the default country and the read only fields.
* Makes the barcode repetitive.
* Removes all user fields from the patron data model.
* Adds the second address, the local code, the source fields in the
  patron data model.
* Closes #1724, #1634, #1615, #1490, #1467, #1318, #1384.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1905?milestone=284341

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* https://github.com/rero/rero-ils-ui/pull/529
* https://github.com/rero/ng-core/pull/354

## How to test?

- check several scenarios of online registration and then create a patron
- create a patron directly
- check existing username, email, etc.

## Code review check list
